metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -237,6 +273,338 @@ def create(cls, config_dict: Dict, ctx: PipelineContext) -> "SQLServerSource":
+            """
+            SELECT DISTINCT
+                *
+            FROM #ProceduresDependencies
+            WHERE referenced_object_type IN ('U ', 'V ', 'P ');
+            """
+        )
+
+        # PAY ATTENTION:
+        # at 19.07.2024 self.get_db_name method converts db name to lowercase forcely:
+        # return str(engine.url.database).strip('"').lower()
+        # to avoid inconsistency, referenced_database_name in this module repeats the same behavior
+        # remove after fixing
+        db_force_converting = "lower()" in inspect_objects.getsource(self.get_db_name)
+
+        for row in _dependencies:
+
+            if row:
+                _key = f"{row['current_db'].lower() if db_force_converting else row['current_db']}.{row['procedure_schema']}.{row['procedure_name']}"
+
+                self.procedures_dependencies[_key].append(
+                    {
+                        "referenced_database_name": row[
+                            "referenced_database_name"
+                        ].lower()
+                        if db_force_converting
+                        else row["referenced_database_name"],
+                        "referenced_schema_name": row["referenced_schema_name"],
+                        "referenced_entity_name": row["referenced_entity_name"],
+                        "referenced_object_type": row["referenced_object_type"],
+                        "is_selected": row["is_selected"],
+                        "is_select_all": row["is_select_all"],
+                        "is_updated": row["is_updated"],
+                    }
+                )
+
+        trans.commit()
+
+    def _populate_object_links(self, conn: Connection, db_name: str) -> None:
+        # see https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-sql-expression-dependencies-transact-sql
+        _links = conn.execute(
+            """
+            SELECT
+              DB_NAME() AS cur_database_name,
+              OBJECT_SCHEMA_NAME(referencing_id) AS dst_schema_name,
+              OBJECT_NAME(referencing_id) AS dst_object_name,
+              so_dst.type AS dst_object_type,
+              referenced_server_name AS src_server_name,
+              referenced_database_name AS src_database_name,
+              COALESCE(OBJECT_SCHEMA_NAME(referenced_id), referenced_schema_name) AS src_schema_name,
+              COALESCE(OBJECT_NAME(referenced_id), referenced_entity_name) AS src_object_name,
+              so_src.type AS src_object_type
+            FROM sys.sql_expression_dependencies
+              LEFT JOIN sys.objects AS so_dst ON so_dst.object_id = referencing_id
+              LEFT JOIN sys.objects AS so_src ON so_src.object_id = referenced_id
+            WHERE so_dst.type in ('U ', 'V ', 'P ')
+              AND so_src.type in ('U ', 'V ', 'P ');
+            """
+        )
+        # {"U ": "USER_TABLE", "V ": "VIEW", "P ": "SQL_STORED_PROCEDURE"}
+        for row in _links:
```

Comment:
do both this query and the other one give store procedure dependencies? bit confused by this